widget/segmentedentry.go

```diff
@@ -0,0 +1,230 @@
+	// a separator after them.
+	DelimitAt []int
+
+	allowedRunes []rune
+	length       int
+
+	entries  []*SingleRuneEntry
+	selected int
+}
+
+// NewSegmentedEntry returns a new SegmentedEntry of the given length allowing
+// the given runes.
+func NewSegmentedEntry(allowedRunes []rune, length int) *SegmentedEntry {
+	// Initialize the input
+	input := &SegmentedEntry{
+		allowedRunes: allowedRunes,
+		length:       length,
+		entries:      make([]*SingleRuneEntry, length),
+		selected:     0,
+	}
+	input.ExtendBaseWidget(input)
+	return input
+}
+
+// GetValue returns the value of all the entries in the grid combined.
+func (g *SegmentedEntry) GetValue() string {
+	var out string
+	for _, e := range g.entries {
+		out += strings.TrimSpace(e.Text)
+	}
+	return out
+}
+
+// FocusGained implements fyne.Focusable
+func (g *SegmentedEntry) FocusGained() {
+}
+
+// FocusLost implements fyne.Focusable
+func (g *SegmentedEntry) FocusLost() {
+}
+
+// FocusIndex will focus the entry at the given index.
+func (g *SegmentedEntry) FocusIndex(idx int) {
+	fyne.CurrentApp().Driver().CanvasForObject(g.entries[idx]).Focus(g.entries[idx])
+	g.selected = idx
+}
+
+// TypedKey forwards the event to the focussed entry
+func (g *SegmentedEntry) TypedKey(key *fyne.KeyEvent) {
+	g.entries[g.selected].TypedKey(key)
+}
+
+// TypedRune forwards the rune to the focussed entry
+func (g *SegmentedEntry) TypedRune(r rune) {
+	g.entries[g.selected].TypedRune(r)
+}
+
+// Disbale disables all entries in the grid.
+func (g *SegmentedEntry) Disable() {
+	for _, e := range g.entries {
+		e.Disable()
+	}
+}
+
+// Enable enables all entries in the grid.
+func (g *SegmentedEntry) Enable() {
+	for _, e := range g.entries {
+		e.Enable()
+	}
+}
+
+// CreateRenderer will initialize the entries and create a renderer
+// for them.
+func (g *SegmentedEntry) CreateRenderer() fyne.WidgetRenderer {
+	g.ExtendBaseWidget(g)
+
+	// Set up the entries
+	for i := 0; i < g.length; i++ {
+		// Create the entry
+		g.entries[i] = NewSingleRuneEntry(g.allowedRunes)
+
+		// Take a reference to this index for the closures
+		this := i
+
+		// Navigation for this entry
+		g.entries[i].OnNavigate = func(key *fyne.KeyEvent) {
+			// This was a regular input
+			if key == nil {
+				if len(g.GetValue()) == g.length && g.OnCompletion != nil {
+					g.OnCompletion(g.GetValue())
+					return
+				}
+				if this != g.length-1 {
+					g.FocusIndex(this + 1)
+				}
+				return
+			}
+			switch key.Name {
+			case fyne.KeyBackspace:
+				if this == 0 {
+					return
+				}
+				if g.entries[this-1].Text != "" {
+					g.entries[this-1].SetText("")
+					g.FocusIndex(this - 1)
+				}
+			case fyne.KeyLeft:
+				if this == 0 {
+					return
+				}
+				g.FocusIndex(this - 1)
+			case fyne.KeyRight:
+				if this == g.length-1 {
+					return
+				}
+				g.FocusIndex(this + 1)
+			}
+		}
+	}
+
+	return &segmentedEntryRenderer{input: g}
+}
+
+func (g *SegmentedEntry) objects() []fyne.CanvasObject {
```

Comment:
Just a suggestion, but it seems like this would be cleaner in the Objects() function of the renderer to save this additional method.